include/cxx.h

```diff
@@ -826,6 +826,87 @@ template <typename T>
 Box<T>::Box(uninit) noexcept {}
 #endif // CXXBRIDGE1_RUST_BOX
 
+#ifndef CXXBRIDGE1_RUST_OPTION
+template <typename T>
+class Option final {
+public:
+  Option() noexcept;
+  Option(Option&&) noexcept;
+  Option(T&&) noexcept;
+  ~Option() noexcept;
+  void drop() noexcept;
+
+  const T *operator->() const;
+  const T &operator*() const;
+  T *operator->();
+  T &operator*();
+
+  bool has_value() const noexcept;
+  const T& value() const;
+
+  union OptionContents {
+    T value;
+    size_t empty;
+
+    OptionContents() {
+      empty = 0;
+    }
+    // Destruction handled in Option since it knows if type was created
+    ~OptionContents() {}
+  };
+  OptionContents inner;  // Avoid initialization
+};
+#endif // CXXBRIDGE1_RUST_OPTION
+
+#ifndef CXXBRIDGE1_RUST_OPTION
+#define CXXBRIDGE1_RUST_OPTION
+template <typename T>
+Option<T>::Option() noexcept {}
+
+template <typename T>
+Option<T>::Option(Option&& other) noexcept {
+  if (other.inner.empty != 0) {
+    inner.value = std::move(other.inner.value);
+  }
+  other.inner.empty = 0;
+}
+
+template <typename T>
+Option<T>::~Option() noexcept {
+  this->drop();
+}
+
+template <typename T>
+const T *Option<T>::operator->() const {
+  return &inner.value;
```

Comment:
Might be nice to add some debug check for `has_value()` where you assume value is set.

Reply:
do you know how to do debug checks portably? I can't tell if I can rely on `NDEBUG`